src/rsz/test/cpp/TestBufferRemoval.cc

```diff
@@ -0,0 +1,225 @@
+    db_network_->setBlock(block);
+    block->setDieArea(odb::Rect(0, 0, 1000, 1000));
+    // register proper callbacks for timer like read_def
+    sta_->postReadDef(block);
+    odb::dbModule* module = odb::dbModule::create(block, "top");
+    odb::dbMaster* bufx1 = db_->findMaster("BUF_X1");
+    odb::dbMaster* bufx2 = db_->findMaster("BUF_X2");
+    odb::dbMaster* bufx4 = db_->findMaster("BUF_X4");
+    odb::dbMaster* bufx8 = db_->findMaster("BUF_X8");
+
+    odb::dbInst* b1 = odb::dbInst::create(block, bufx1, "b1", module);
+    b1->setSourceType(odb::dbSourceType::TIMING);
+    b1->setLocation(100, 100);
+    b1->setPlacementStatus(odb::dbPlacementStatus::PLACED);
+
+    odb::dbInst* b2 = odb::dbInst::create(block, bufx2, "b2", module);
+    b2->setSourceType(odb::dbSourceType::TIMING);
+    b2->setLocation(200, 200);
+    b2->setPlacementStatus(odb::dbPlacementStatus::PLACED);
+
+    odb::dbInst* b3 = odb::dbInst::create(block, bufx4, "b3", module);
+    b3->setSourceType(odb::dbSourceType::TIMING);
+    b3->setLocation(300, 300);
+    b3->setPlacementStatus(odb::dbPlacementStatus::PLACED);
+
+    odb::dbInst* b4 = odb::dbInst::create(block, bufx8, "b4", module);
+    b4->setSourceType(odb::dbSourceType::TIMING);
+    b4->setLocation(400, 400);
+    b4->setPlacementStatus(odb::dbPlacementStatus::PLACED);
+
+    odb::dbNet* in1 = odb::dbNet::create(block, "in1");
+    odb::dbNet* n1 = odb::dbNet::create(block, "n1");
+    odb::dbNet* n2 = odb::dbNet::create(block, "n2");
+    odb::dbNet* n3 = odb::dbNet::create(block, "n3");
+    odb::dbNet* out1 = odb::dbNet::create(block, "out1");
+
+    odb::dbWire::create(in1);
+    odb::dbWire::create(n1);
+    odb::dbWire::create(n2);
+    odb::dbWire::create(n3);
+    odb::dbWire::create(out1);
+
+    in1->setSigType(odb::dbSigType::SIGNAL);
+    n1->setSigType(odb::dbSigType::SIGNAL);
+    n2->setSigType(odb::dbSigType::SIGNAL);
+    n3->setSigType(odb::dbSigType::SIGNAL);
+    out1->setSigType(odb::dbSigType::SIGNAL);
+
+    odb::dbBTerm* inPort = odb::dbBTerm::create(in1, "in1");
+    inPort->setIoType(odb::dbIoType::INPUT);
+    odb::dbBPin* inPin = odb::dbBPin::create(inPort);
+    odb::dbTechLayer* layer = block->getTech()->findRoutingLayer(0);
+    odb::dbBox::create(inPin, layer, 0, 0, 10, 10);
+    inPin->setPlacementStatus(odb::dbPlacementStatus::FIRM);
+    // this is needed to create top level STA port object
+    sta_->makePortPin("in1", "input");
+
+    odb::dbBTerm* outPort = odb::dbBTerm::create(out1, "out1");
+    outPort->setIoType(odb::dbIoType::OUTPUT);
+    odb::dbBPin* outPin = odb::dbBPin::create(outPort);
+    layer = block->getTech()->findRoutingLayer(0);
+    odb::dbBox::create(outPin, layer, 990, 990, 1000, 1000);
+    outPin->setPlacementStatus(odb::dbPlacementStatus::FIRM);
+    // why is this step of creating STA port object not automatic after creating
+    // bterm and bpin?
+    sta_->makePortPin("out1", "output");
```

Comment:
@maliberty Do we need a dbsta callback for this?

Reply:
Actually, this is not needed anymore after adding sta_->postReadDef(block) callback.   The codes have been removed.